tools/src/test/java/org/terracotta/utilities/io/FilesTestBase.java

```diff
@@ -871,24 +884,68 @@ public boolean mirrors(PathDetails that, boolean compareAttributes) {
-              && Objects.equals(this.isReadOnly, that.isReadOnly)
-              && Objects.equals(this.isSystem, that.isSystem)
-              && Objects.equals(this.isHidden, that.isHidden)
-              && Objects.equals(this.isArchive, that.isArchive)
-              && Objects.equals(this.permissions, that.permissions);
+
+          // If both files reside in the same type of file store, compare the last modified time;
+          // if in different file stores, can't make reliable assertions regarding time.
+          boolean lastModifiedTimeEqual;
+          if (this.fileStoreType.equals(that.fileStoreType)) {
+            // Windows, at least, doesn't copy lastModifiedTime for non-files
+            lastModifiedTimeEqual =
+                !this.types.contains(FileType.FILE) || Objects.equals(this.lastModifiedTime, that.lastModifiedTime);
+          } else {
+            LOGGER.trace("Skipping time comparison: {} != {}", this.fileStoreType, that.fileStoreType);
+            lastModifiedTimeEqual = true;   // Don't fail if file stores are different
+          }
+
+          // If both files reside in a file store supporting "dos", compare DOS attributed
+          boolean dosAttributesEqual;
+          if (this.attributeViews.contains("dos") && that.attributeViews.contains("dos")) {
+            dosAttributesEqual = Objects.equals(this.isReadOnly, that.isReadOnly)
+                && Objects.equals(this.isSystem, that.isSystem)
+                && Objects.equals(this.isHidden, that.isHidden)
+                && Objects.equals(this.isArchive, that.isArchive);
+          } else {
+            LOGGER.trace("Skipping 'dos' attributes comparison: {} / {}", this.attributeViews, that.attributeViews);
+            dosAttributesEqual = true;    // Don't fail if file stores are different
+          }
+
+          // If both files reside in a file store supporting "posix", compare Posix attributes
+          boolean posixAttributesEqual;
+          if (this.attributeViews.contains("posix") && that.attributeViews.contains("posix")) {
+            posixAttributesEqual = Objects.equals(this.permissions, that.permissions);
+          } else {
+            LOGGER.trace("Skipping 'posix' attributes comparison: {} / {}", this.attributeViews, that.attributeViews);
+            posixAttributesEqual = true;    // Don't fail if file stores are different
+          }
+
+          return lastModifiedTimeEqual && dosAttributesEqual && posixAttributesEqual;
         }
       }
 
       return true;
     }
 
+    /**
+     * Determine what {@code FileAttributeView} types are supported for a path.
+     *
+     * @param fullPath  the fully-qualified path to interrogate
+     * @param fileStore the {@code FileStore} holding {@code fullPath}
+     * @return the set of supported views
+     */
+    private Set<String> supportedAttributeViews(Path fullPath, FileStore fileStore) {
```

Comment:
I think this is one of those cases where a stream expression might communicate the intent better here:
```java
return fullPath.getFileSystem(().supportedFileAttributeViews().stream()
    .filter(fileStore::supportsFileAttributeView)
    .collect(collectingAndThen(toSet(), Collections::unmodifiableSet))
```

Reply:
I also wonder if it would be better if you double fetched the FileStore... to make clean the relationship between these two parameters in a "legal" call.

Reply:
The reason it's split like this is because, if `fullPath` refers to a non-existing file, `getFileStore` throws -- that's why it's determined from the root in the caller.

As for the stream expression -- I can see its clarity.